bernas670/sinf
# Auth backend and connection with frontend

# Login, Logout and Session System is working.

Users that have an account can now log into the system, surf the website and logout. There is also a session mechanism using the localStorage that keeps an auth-token.

# How to test?

* First you will need to get the server up and running. Go to the server folder, install all dependencies using ```yarn install```, run ```yarn run migration:run``` and ```yarn run start:watch```. The server should be up on port 8000.

![image](https://user-images.githubusercontent.com/44146077/101297670-0d699980-3822-11eb-89aa-126bcce5c82f.png)

* Go to the client's folder and install all dependencies using ```yarn install```, then ```yarn run build``` and ```yarn run start```. The client should be running on port 3000.

![image](https://user-images.githubusercontent.com/44146077/101297675-13f81100-3822-11eb-89e2-300f2ab72849.png)

* Go to ```http://localhost:3000``` on your browser and access the site.

![image](https://user-images.githubusercontent.com/44146077/101297678-18bcc500-3822-11eb-982c-a2450ae544ef.png)
![image](https://user-images.githubusercontent.com/44146077/101297681-1bb7b580-3822-11eb-8ad3-6a0ca87f39a3.png)


# What to test?

* Log in with a valid user. There are five valid user's on the migration script.
* Log in with an invalid user. That user shouldn't be able to acces the site.
* Try to access web pages without being logged in (it shouldn't work).
* Log out of the site using the logout button.
* Look for bugs.
